src/boot/mexpr.ml

```diff
@@ -481,29 +493,35 @@ let rec debruijn env t =
      -> TmUtest(fi,debruijn env t1,debruijn env t2,debruijn env tnext)
 
 
-
 let rec tryMatch env value pat =
   let go v p env = Option.bind (fun env -> tryMatch env v p) env in
-  let rec splitNth n = function [] -> ([],[]) | x::xs ->
-    if n > 0 then let (pre,post) = splitNth (n-1) xs in (x::pre,post) else ([],x::xs) in
+  let splitNth n s =
+    assert (n <= Mseq.length s);
+    if Mseq.length s == 0 then (Mseq.empty, Mseq.empty)
+    else if n < Mseq.length s then Mseq.split_at s n
+    else (s, Mseq.empty)
```

Comment:
Why are these special cases needed? Is our intended semantics of `split_at` different from the one in the library?

Reply:
Yes, this local function `slitNth` has a different semantics. Less so if we change the semantics of `split_at` so that `split_at [1,2,3] 3` equals `([1,2,3],[])`. Still `splitNth s 0` equals `([],[])` which is not true for `split_at` and I don't think we want `split_at` to have this semantics. In any case, I agree that we should depart from the `split_at [1,2,3] 3` semantics of batteries and change the code a bit to remove `splitNth` altogether.

Reply:
I decided to keep `splitNth` but change its name to reflect the difference in semantics.